metrics/ci_integrations/lib/jenkins/adapter/jenkins_ci_client_adapter.dart

```diff
@@ -63,10 +66,12 @@ class JenkinsCiClientAdapter implements CiClient {
   /// The [limits] can be used to set a range-specifier for the request.
   Future<JenkinsBuildingJob> _fetchBuilds(
     String projectId, {
-    JenkinsQueryLimits limits = const JenkinsQueryLimits.empty(),
+    JenkinsQueryLimits limits,
   }) async {
-    final newBuildsFetchResult =
-        await jenkinsClient.fetchBuilds(projectId, limits: limits);
+    final newBuildsFetchResult = await jenkinsClient.fetchBuilds(
+      projectId,
+      limits: limits ?? JenkinsQueryLimits.endBefore(jenkinsBuildsFetchLimit),
```

Comment:
I think we should move limit to to fetchBuilds() when we call _fetchBuilds() as that will make our intent on limiting the data to fetchAll function (for now)